conda/core/path_actions.py

```diff
@@ -1042,3 +1042,34 @@ def target_full_path(self):
     def __str__(self):
         return ('ExtractPackageAction<source_full_path=%r, target_full_path=%r>'
                 % (self.source_full_path, self.target_full_path))
+
+
+class DumpRecordAction(PathAction):
+
+    def __init__(self, target_pkgs_dir, target_extracted_dirname, record):
+        self.target_pkgs_dir = target_pkgs_dir
+        self.target_extracted_dirname = target_extracted_dirname
+        self.record = record
+
+    def verify(self):
+        self._verified = True
+
+    def execute(self):
+
+        meta = join(self.target_full_path, 'info', 'record.json')
+        log.trace("writing record to %s", meta)
+        with open(meta, 'w') as out:
```

Comment:
You should just use `self.target_full_path` here.

Also, I'm not yet 100% comfortable with `record.json` as a file name.  I'd probably be fine with `repodata_record.json` as a filename.  Naming is hard, and also can't be easily changed once it's set.  So let's make sure we're all as comfortable with it as we can be.  I'm open to more discussion.